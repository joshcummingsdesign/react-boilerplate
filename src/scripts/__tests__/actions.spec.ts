import * as moxios from 'moxios';
import { setSearchTerm, addApiData, getApiDataThunk } from '../actions';
const image = require('../__mocks__/images.json')[0];

test('setSearchTerm', () => {
  expect(setSearchTerm('qui')).toMatchSnapshot();
});

test('addApiData', () => {
  expect(addApiData(image)).toMatchSnapshot();
});

test('getApiDataThunk', done => {
  const dispatchMock = jest.fn();
  moxios.withMock(() => {
    getApiDataThunk(image.id)(dispatchMock);
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request
        .respondWith({
          status: 200,
          response: image
        })
        .then(() => {
          expect(request.url).toEqual(`https://jsonplaceholder.typicode.com/photos/${image.id}`);
          expect(dispatchMock).toBeCalledWith(addApiData(image));
          done();
        });
    });
  });
});
