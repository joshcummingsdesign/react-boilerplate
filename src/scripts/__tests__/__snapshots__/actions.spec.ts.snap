// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addApiData 1`] = `
Object {
  "payload": Object {
    "albumId": 1,
    "id": 1,
    "thumbnailUrl": "https://via.placeholder.com/150/92c952",
    "title": "accusamus beatae ad facilis cum similique qui sunt",
    "url": "https://via.placeholder.com/600/92c952",
  },
  "type": "ADD_API_DATA",
}
`;

exports[`setSearchTerm 1`] = `
Object {
  "payload": "qui",
  "type": "SET_SEARCH_TERM",
}
`;
